src/main/java/servlets/LoginServlet.java

```diff
@@ -53,17 +47,18 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   /** Build the OAuth consent page redirect URL. */
   private String getOAuthRedirectURL() {
-    String redirectUri = REDIRECT_URI + getRedirectUri();
-    
-    return String.format("%s?%s&%s&%s&%s", OAUTH_LOGIN_URI, CLIENT_ID,
-      RESPONSE_TYPE, redirectUri, SCOPE); 
+    return String.format("%s?%s&%s&%s&%s", OAuthConstants.OAUTH_LOGIN_URI, 
+      OAuthConstants.CLIENT_ID, OAuthConstants.RESPONSE_TYPE, getRedirectUri(), 
+      OAuthConstants.SCOPE); 
   }
 
   // Build a valid redirect URI to the OAuthCallbackServlet.
   private String getRedirectUri() {
     try {
-      URI domainUri = URI.create(System.getenv().get(ENVIRONMENT_VARIABLE));
-      return domainUri.resolve(OAUTH_CALLBACK_SERVLET).toString();
+      URI domainUri = URI.create(System.getenv().get(
```

Comment:
Is this to get things for the URI? Just a little confusion.

Reply:
This gets the redirect URI that is saved in the environment variable.

Reply:
Thanks for the clarifying comment!